### DISCUSSION TITLE
Не работает http . все ссылки https работают нормально.

### DISCUSSION THREAD

**Discussion by jura1243** [hufrea/byedpi#104] [General]
на сервере запустил ./ciadpi --disorder 1 --auto=torst --tlsrec 1+s &
на винде через панель управления задал всем использовать сокс прокси.
все работает. не открываются ссылки http . например http://forum.tanki.su/
что можно сделать в этой ситуации?

**Comment by hufrea** [comment]
`disorder` может ломать HTTP. Для него обход можно выключить:
`--proto http --auto=none --disorder 1  --auto=torst --tlsrec 1+s`

**Reply by jura1243** [reply]
заработало, спасибо.

**Comment by radioxoma** [comment]
FYI, ArchLinux, МГТС, мособласть. `ciadpi --disorder 1` ломает https://lk.megafon.ru, а `ciadpi --split 1` нет.

**Reply by zimbabwe303** [reply]
Чисто из моего опыта:
1. Начинаем строку с самого мягкого варианта, т.е. `--split 1`
2. Затем ставим какой-то `--auto`, что означает "если не срабатывает первое, то идём дальше", в моём случае это `--auto r` (redirect), т.к. при попытке зайти, например, на rutracker.org я получаю страничку от провайдера (заблокировано по решению властей), а это значит, что происходит на неё редирект, и на него можно среагировать. Если вместо странички получаете просто ошибку браузера, то нужен `--auto t`, `--auto s` или `--auto t,s` (можно попробовать и другие комбинации).
3. Затем уже ставим `--disorder 1`, `--oob 1` или `--disoob 1`: предполагаю, что т.к. дисордер сработает уже после первой реакции вундервафли, пакет возвращается целым, а вундервафлю обманывает.
4. Затем ставим `--auto t`: если дисордер заканчивается RST, то далее идёт финальная стадия с фейковыми пакетами и сдвижкой TLS.
5. Ставим `--fake -1 --tls 1+s`: для всех нормальных TLS рукопожатий этот мусор поломает коннект (поэтому его и нельзя использовать отдельно или в начале), но если сначала вторгается вундервафля со своей MITM атакой, то ломается вундервафля, заглатывая мусор. В итоге заходит, например, на РуТрекер, Twirpx или ещё куда.

**Reply by zimbabwe303** [reply]
Ещё, если поток с YouTube прерывается через какое-то время, стоит попробовать двухсложные комбинации типа `--split 1 --oob 1` или сложные смещения типа `--split 1+h` или `--split -1`, но последние сильно снижают скорость и увеличивают вероятность сбоев рукопожатия.
